index.js

```diff
@@ -0,0 +1,563 @@
+  // get the total for the purchase_units.
+  const {total, currencies} = getTotalCost({order});
+  const sameAmount = total.isEqualTo(expectedAmount.value);
+  if(!sameAmount) {
+    throw new BedrockError(
+      `Unexpected amount.`,
+      Errors.Data,
+      {payPalId: order.id, expectedAmount, actualAmount: total.toString()}
+    );
+  }
+  const sameCurrency = currencies.has(expectedAmount.currency_code);
+  if(!sameCurrency) {
+    throw new BedrockError(
+      'Unexpected currency.',
+      Errors.Data,
+      {
+        public: true,
+        payPalId: order.id,
+        currencyCode: expectedAmount.currency_code
+      }
+    );
+  }
+  return true;
+};
+
+/**
+ * Checks that the existing PayPal order is there
+ * and match the expectedAmount.
+ *
+ * @param {object} options - Options to use.
+ * @param {object} options.payment - A Bedrock payment.
+ * @param {object} options.amount - A Paypal amount object.
+ * @param {BigNumber|number|string} options.expectedAmount - How is expected.
+ *
+ * @returns {object} The updated paypal order.
+ */
+const updateGatewayPaymentAmount = async (
+  {pendingPayment, updatedPayment}) => {
+  const expectedAmount = {
+    currency_code: pendingPayment.currency,
+    value: pendingPayment.amount
+  };
+  const amount = {
+    currency_code: updatedPayment.currency,
+    value: updatedPayment.amount
+  };
+  // ensure the order is still there.
+  const order = await getOrderFromPayment({payment: pendingPayment});
+  const value = formatAmount({amount});
+  compareAmount({order, expectedAmount});
+  const patch = [{
+    op: 'replace',
+    path: `/purchase_units/@reference_id=='${pendingPayment.id}'/amount`,
+    value
+  }];
+  const updatedOrder = await updateOrder({order, patch});
+  // make sure the swap occurred.
+  compareAmount({order: updatedOrder, expectedAmount: amount});
+  const payment = Object.assign(pendingPayment, updatedPayment);
+  return {updatedOrder, payment};
+};
+
+/**
+ * This first checks to ensure the client's order matches
+ * a valid PayPal Order.
+ *
+ * @param {object} options - Options to use.
+ * @param {Array<object>} options.clientPurchases - Client purchases.
+ * @param {Array<object>} options.paypalPurchases - PayPal purchases.
+ *
+ * @returns {object} A verified purchase.
+ */
+const verifyPurchase = ({paypalPurchases}) => {
+  // PayPal currently only supports one purchase_item at a time.
+  // Hence we use the first purchase_item.
+  // This will need to be refactored if we allow a user
+  // to purchase more than one product in an order or PayPal allows
+  // more than one purchase_unit per order.
+  //
+  if(!paypalPurchases) {
+    throw new BedrockError('Missing PayPal purchases.', Errors.Data);
+  }
+  const [paypalPurchase] = paypalPurchases;
+  if(!paypalPurchase) {
+    throw new BedrockError('Missing PayPal purchase.', Errors.Data);
+  }
+  // throw an error if the client altered their purchase amount
+  // after the transaction succeeded.
```

Comment:
Not sure what these comments mean.

Reply:
they were comments for a line which was removed. my apologies removing it. https://github.com/digitalbazaar/bedrock-payment-paypal/pull/2/commits/42a81daf80556f5ef82c3744e34cb7533e466d79